src/System.Management.Automation/engine/parser/Parser.cs

```diff
@@ -1629,6 +1630,20 @@ private ScriptBlockAst NamedBlockListRule(Token lCurly, List<UsingStatementAst>
                 {
                     default:
                         UngetToken(blockNameToken);
+                        extent = ExtentOf(startExtent, endExtent);
+
+                        // If 'lCurly' is null, then handle the next token in 'CompleteScriptBlockBody'.
+                        if (lCurly == null) { goto finished_named_block_list; }
```

Comment:
This feels wrong - I'd expect the same error in a ps1/psm1 file as in a function/anonymous script block.

Reply:
Fixed. But now it seems to me that this code block in `CompleteScriptBlockBody` will never be hit:
https://github.com/PowerShell/PowerShell/blob/cdc2015418e92f321311587d61570648ca392038/src/System.Management.Automation/engine/parser/Parser.cs#L1548-L1555

What shall we do with this code block?

Reply:
I take that back. This block of code still can be hit when `lCurly == null` and `nextToken.Kind == RCurly`.